ls1intum/ArTEMiS
# Docker setup for development environment

### Checklist
- [x] I run `yarn run webpack:build:main`: the project builds without errors. (not necessary)
- [x] I run `yarn lint`: the project builds without code style warnings. (not necessary)
- [x]  I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de. (not necessary)
- [x] I added (end-to-end) test cases for the new functionality. (not necessary)

### Motivation and Context
At the moment it is quite hard to setup the local development environment especially due to outdated dependencies. Using docker-compose will help with onboarding new developers.

### Description
A basic docker setup using docker-compose to provide a local development environment. Currently frontend and server are running in the same container. The project specific gradle wrapper is used to run the project in the container via `./gradlew bootRun`.

The setup can be used by running `docker-compose up`. Maybe you have to stop your local mysql service (`service mysql stop`) to prevent port conflicts.

One can get a shell to the app container via `docker exec -it $(docker-compose ps -q artemis-app) sh` and a shell to the mysql container via `docker exec -it $(docker-compose ps -q artemis-mysql) mysql`.

Dependencies:

* [docker](https://docs.docker.com/install/)
* [docker-compose](https://docs.docker.com/compose/install/)

### Steps for Testing
Just run `docker-compose up`. The setup should not have any effect on the rest of the application. 

### Screenshots
